### ISSUE TITLE
Implement HeartDiseaseModel class defining the core CNN architecture

### ISSUE THREAD

**Issue by QbaTa** [Artificial-Intelligence-in-Medicine-AGH/cnn-classifier#26]
### Description

Create a new file `Model.py` that defines **the main CNN model architecture** for heart disease image classification.
This class `HeartDiseaseModel` will inherit from `torch.nn.Module` and form the **heart of the project**, defining how input images are processed by convolutional and fully connected layers to produce predictions.

### Class Signature
```python
class HeartDiseaseModel(nn.Module):
    def __init__(self, num_classes: int = 2):
        super(HeartDiseaseModel, self).__init__()
        ...
    
    def forward(self, x: torch.Tensor) -> torch.Tensor:
        ...

``` 

### Goals

Implement full CNN architecture including:

- Convolutional layers (`nn.Conv2d`)
- Activation layers (`nn.ReLU`)
- Pooling layers (`nn.MaxPool2d`)
- Flattening operation (`nn.Flatten`)
- Fully connected (`nn.Linear`) layers producing class logits

### Implementation Notes

File `Model.py`
Class `HeartDiseaseModel`
